runtime/shared_common/CacheMap.cpp

```diff
@@ -2010,15 +2176,26 @@ SH_CacheMap::commitROMClass(J9VMThread* currentThread, ShcItem* itemInCache, SH_
 	}
 
 	srcwInCache = (ScopedROMClassWrapper*) ITEMDATA(itemInCache); /* must be after name is written */
-	srcw.theCpOffset = (J9SRP)((BlockPtr) cpw - (BlockPtr) srcwInCache);
-	/*Calc the J9SRP*/
-	srcw.romClassOffset = (J9SRP)((BlockPtr) romClassBuffer - (BlockPtr) srcwInCache);
+
+	getJ9ShrOffsetFromAddress(cpw, &srcw.theCpOffset);
+	/* Calculate the J9ShrOffset */
+	getJ9ShrOffsetFromAddress(romClassBuffer, &srcw.romClassOffset);
 	if (useScope) {
 		const J9UTF8* mcToUse = modContextInCache;
 		const J9UTF8* ptToUse = partitionInCache;
 
-		srcw.modContextOffset = (J9SRP)((modContextInCache) ? ((BlockPtr) mcToUse - (BlockPtr) srcwInCache) : 0);
-		srcw.partitionOffset = (J9SRP)((partitionInCache) ? ((BlockPtr) ptToUse - (BlockPtr) srcwInCache) : 0);
+		if (NULL == modContextInCache) {
+			J9ShrOffSet offset = {0};
+			srcw.modContextOffset = offset;
```

Comment:
This isn't valid, the stack allocated `J9ShrOffSet offset` is out of scope and can be overwritten after this block exits.

Reply:
Fixed